gov2/iam/common/examples_test.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+)
+
+func TestExamples(t *testing.T) {
+
+	cfg, err := config.LoadDefaultConfig(context.Background())
+
+	if err != nil {
+		panic("Couldn't load a configuration")
+	}
+	iamClient := iam.NewFromConfig(cfg)
+
+	var created ExampleCreatedResources
+
+	func() {
+		defer func() {
+			if err := recover(); err != nil {
+				// Whoops. Clean up whatever we can.
+				t.Logf("Failed to run tests: %v", err)
+
+				// Try and clean up what we can.
+
+				_, xerr := iamClient.DeleteUser(context.Background(), &iam.DeleteUserInput{UserName: aws.String(ExampleUserName)})
+				if xerr != nil {
+					t.Error("Couldn't clean up user")
+				}
+				_, xerr = iamClient.DetachRolePolicy(context.Background(), &iam.DetachRolePolicyInput{
+					PolicyArn: aws.String(ExamplePolicyARN),
+					RoleName:  aws.String(ExampleRoleName),
+				})
+				if xerr == nil {
+					t.Log("Cleaned up role policy attach")
+				} else {
+					t.Log("Failed to detach role policy")
+				}
+				_, xerr = iamClient.DeleteRole(context.TODO(), &iam.DeleteRoleInput{RoleName: aws.String(ExampleRoleName)})
+
+				if xerr == nil {
+					t.Log("Cleaned up role")
+				} else {
+					t.Log("Failed to clean up role")
+				}
+
+				// get the caller identity
+				stsClient := sts.NewFromConfig(cfg)
+				identity, xerr := stsClient.GetCallerIdentity(context.Background(), &sts.GetCallerIdentityInput{})
+
+				if xerr != nil {
+					panic("Something terrible has happened.")
```

Comment:
Is it possible to be a bit more specific with the error message? Is it something that can be resolved by the user doing something differently? Or is it a matter of trying to run the example again?